thrust/thrust/zip_function.h

```diff
@@ -181,6 +170,12 @@ class zip_function
 
 #  endif // _CCCL_STD_VER
 
+  //! Returns a reference to the underlying function.
+  _CCCL_HOST_DEVICE Function& underlying_function() const
+  {
+    return func;
+  }
```

Comment:
Thanks for adding this. This makes it easier to "undo" the zip_function when the zip iterator has been destructured, and will be really helpful for the `thrust::transform` performance work.